src/c++/perf_analyzer/request_rate_manager.cc

```diff
@@ -196,24 +198,45 @@ RequestRateManager::PauseWorkers()
   // Pause all the threads
   execute_ = false;
 
+  // Wait to see all threads are paused.
+  for (auto& thread_config : threads_config_) {
+    while (!thread_config->is_paused_) {
+      std::this_thread::sleep_for(std::chrono::milliseconds(10));
+    }
+  }
+}
+
+void
+RequestRateManager::ConfigureThreads()
+{
   if (threads_.empty()) {
-    while (threads_.size() < max_threads_) {
+    size_t num_of_threads = DetermineNumThreads();
+    // max_threads_;
+    // if (on_sequence_model_) {
+    //   num_of_threads = std::min(max_threads_, num_of_sequences_);
+    // }
+    while (threads_.size() < num_of_threads) {
       // Launch new thread for inferencing
       threads_stat_.emplace_back(new ThreadStat());
       threads_config_.emplace_back(
-          new RequestRateWorker::ThreadConfig(threads_.size(), max_threads_));
+          new RequestRateWorker::ThreadConfig(threads_.size(), num_of_threads));
 
       workers_.push_back(
           MakeWorker(threads_stat_.back(), threads_config_.back()));
 
       threads_.emplace_back(&IWorker::Infer, workers_.back());
-    }
-  }
 
-  // Wait to see all threads are paused.
-  for (auto& thread_config : threads_config_) {
-    while (!thread_config->is_paused_) {
-      std::this_thread::sleep_for(std::chrono::milliseconds(10));
+      // Compute the number of sequences for each thread (take floor)
```

Comment:
This code looks like it is inside of the while loop. Am i reading that correctly? If so, that seems wrong.